### DISCUSSION TITLE
To get access policy

### DISCUSSION THREAD

**Discussion by Opperessor** [sqlalchemy/sqlalchemy#9432] [Usage Questions]
Is there a way that we can get the table names that the user can access or has permission to access to using sqlalchemy (bigquery).

**Comment by zzzeek** [comment]
this would be via an inspection method called get_table_names(): https://docs.sqlalchemy.org/en/20/core/reflection.html#sqlalchemy.engine.reflection.Inspector.get_table_names

e.g.

```py
from sqlalchemy import create_engine, inspect
engine = create_engine("...")

inspector = inspect(engine)
print(inspector.get_table_names())
```

however, we've heard that the bigquery driver might have problems supporting this.  So you'd want to get further help from the developers of the bigquery driver over at https://github.com/googleapis/python-bigquery-sqlalchemy, the existing problem we heard about is at https://github.com/googleapis/python-bigquery-sqlalchemy/issues/838
